tests/ui/spirv-attr/matrix-type.rs

```diff
@@ -0,0 +1,43 @@
+// build-pass
+// compile-flags: -Ctarget-feature=+RayTracingKHR,+ext:SPV_KHR_ray_tracing
+
+use spirv_std as _;
+
+#[derive(Clone, Copy)]
+#[spirv(matrix)]
+pub struct Affine3 {
+    pub x: glam::Vec3,
+    pub y: glam::Vec3,
+    pub z: glam::Vec3,
+    pub w: glam::Vec3,
+}
+
+impl Affine3 {
+    pub const ZERO: Self = Self {
+        x: glam::Vec3::ZERO,
+        y: glam::Vec3::ZERO,
+        z: glam::Vec3::ZERO,
+        w: glam::Vec3::ZERO,
+    };
+
+    pub const IDENTITY: Self = Self {
+        x: glam::Vec3::X,
+        y: glam::Vec3::Y,
+        z: glam::Vec3::Z,
+        w: glam::Vec3::ZERO,
+    };
+}
+
+impl Default for Affine3 {
+    #[inline]
+    fn default() -> Self {
+        Self::IDENTITY
+    }
+}
+
+#[spirv(closest_hit)]
+pub fn main(
+    #[spirv(object_to_world)] _object_to_world: Affine3,
+    #[spirv(world_to_object)] _world_to_object: Affine3,
+) {
+}
```

Comment:
I believe this isn't actually testing much due to the majority of it being dead code. (Also, nit, the name `Affine3` is a little misleading - it's not an affine transformation - but isn't super important for a test, haha)

I would like to see field accesses/etc. tested as well, I'm nervous about just crossing our fingers and hoping that matricies behave exactly like structs in all ways and no instructions need to be modified to handle matricies specially.

Reply:
Oops, I chose `Affine3` because `glam::f32::Affine3A` has the same layout https://docs.rs/glam/0.18.0/glam/f32/struct.Affine3A.html (Sry, I don't know about math 😨).

I feel adding more tests for field operations in `tests/ui/spirv-attr/* ` is not appropriate because it may should contain only about attr tests.
Is it OK? or is there any better place?

Reply:
Oh! whoops, misread it as 4 Vec4s, not 4 Vec3s, 4 Vec3s definitely _is_ an affine transform, haha, sorry

Yeah, not totally sure about where to put tests, anywhere is probably fine

Reply:
I added tests `to matrix-type.rs` I think this is enough but are there suggestions for more cases?